light_curves/code_src/panstarrs_functions.py

```diff
@@ -4,7 +4,7 @@
 import lsdb
 from dask.distributed import Client
 from data_structures import MultiIndexDFObject
-
+from upath import UPath
```

Comment:
Where does this coming from? Do we really need the extra dependency? (upath is not a direct dependency until this point)

Reply:
I think lsdb.read_hats() will be able to deal with raw s3 strings, so no need to wrap it up as a path object, directly use the urls

Reply:
This is now required by `lsdb`. Sending a raw string for s3 now throws an error:

```
FileNotFoundError: Failed to read HATS at location s3://stpubdata/panstarrs/ps1/public/hats/otmo
```

It also will not accept a filesystem object, only a `UPath`.

Reply:
Anyway, I think it's an upstream issue, their API should be user friendly enough to accept reasonable inputs like an s3 uri string

Reply:
Oh that is interesting. I had heard awhile back that this change was coming (requiring a `UPath`), so maybe a version discrepancy? I also tried locally and got that error with lsdb v0.4.5 (latest).

I agree, it would be nice for them to handle this under the hood for simple simple cases like this where no credentials are required.

Reply:
I'll open an issue. I could just submit a PR but I have too many other pressing tasks at the moment, so maybe in a month or so.

Reply:
I'm slack DM-ing you to debug

Reply:
OK, so conclusion, currently anonym access is not the default and universal_pathlib is required to enable anonym s3 access for `read_hats`. We open issues for this default to change upstream, but in the meantime what you did here is the right approach.

(and it all worked for me locally as I have AWS credentials set up, and boot automatically picked those up, but we don't have it on fornax, thus the users will run into the same FileNotFoundError Troy mentioned).

Reply:
the one todo here thus is to add `universal_pathlib` to the requirements file as we directly use it (although it's already an indirect requirement via hats/lsdb)

Reply:
Opened https://github.com/astronomy-commons/hats/issues/462

Reply:
It hasn't been released though, so I would just open a reminder issue to do the cleanup once we have it installed.

Reply:
Oh right, thanks. I see you opened #383.